From: Martin Schwidefsky <schwidefsky@de.ibm.com>
Subject: IBM Codedrop 2006-01-09 - linux-2.6.15-s390-01-05-october2005.diff

[patch] multiple subchannel sets support.

Add support for multiple subchannel sets. Works with arbitrary devices
in subchannel set 1 and is transparent to device drivers. Although
currently only two subchannel sets are available, this will work
with the architectured maximum number of subchannel sets as well.

Acked-by: Hannes Reinecke <hare@suse.de>

---

 drivers/s390/cio/blacklist.c     |  236 ++++++++++++-------
 drivers/s390/cio/blacklist.h     |    2 
 drivers/s390/cio/chsc.c          |  473 ++++++++++++++++++++++-----------------
 drivers/s390/cio/chsc.h          |    9 
 drivers/s390/cio/cio.c           |  166 +++++++------
 drivers/s390/cio/cio.h           |   11 
 drivers/s390/cio/cmf.c           |    8 
 drivers/s390/cio/css.c           |  256 +++++++++++++--------
 drivers/s390/cio/css.h           |   41 ++-
 drivers/s390/cio/device.c        |   45 ++-
 drivers/s390/cio/device.h        |    1 
 drivers/s390/cio/device_fsm.c    |   29 +-
 drivers/s390/cio/device_id.c     |   24 +
 drivers/s390/cio/device_ops.c    |    4 
 drivers/s390/cio/device_pgid.c   |   43 +--
 drivers/s390/cio/device_status.c |   14 -
 drivers/s390/cio/ioasm.h         |   84 ++++--
 drivers/s390/cio/qdio.c          |  158 +++++++------
 drivers/s390/cio/qdio.h          |   26 +-
 drivers/s390/cio/schid.h         |   26 ++
 drivers/s390/s390mach.c          |   56 +++-
 21 files changed, 1060 insertions(+), 652 deletions(-)

diff -urpN linux-2.6/drivers/s390/cio/blacklist.c linux-2.6-patched/drivers/s390/cio/blacklist.c
--- linux-2.6/drivers/s390/cio/blacklist.c	2006-01-03 04:21:10.000000000 +0100
+++ linux-2.6-patched/drivers/s390/cio/blacklist.c	2006-01-09 15:40:30.000000000 +0100
@@ -1,7 +1,7 @@
 /*
  *  drivers/s390/cio/blacklist.c
  *   S/390 common I/O routines -- blacklisting of specific devices
- *   $Revision: 1.39 $
+ *   $Revision: 1.40 $
  *
  *    Copyright (C) 1999-2002 IBM Deutschland Entwicklung GmbH,
  *			      IBM Corporation
diff -urpN linux-2.6/drivers/s390/cio/cmf.c linux-2.6-patched/drivers/s390/cio/cmf.c
--- linux-2.6/drivers/s390/cio/cmf.c	2006-01-03 04:21:10.000000000 +0100
+++ linux-2.6-patched/drivers/s390/cio/cmf.c	2006-01-09 15:40:30.000000000 +0100
@@ -1,5 +1,5 @@
 /*
- * linux/drivers/s390/cio/cmf.c ($Revision: 1.19 $)
+ * linux/drivers/s390/cio/cmf.c ($Revision: 1.20.2.1 $)
  *
  * Linux on zSeries Channel Measurement Facility support
  *
diff -urpN linux-2.6/drivers/s390/cio/css.c linux-2.6-patched/drivers/s390/cio/css.c
--- linux-2.6/drivers/s390/cio/css.c	2006-01-09 15:40:16.000000000 +0100
+++ linux-2.6-patched/drivers/s390/cio/css.c	2006-01-09 15:40:30.000000000 +0100
@@ -1,7 +1,7 @@
 /*
  *  drivers/s390/cio/css.c
  *  driver for channel subsystem
- *   $Revision: 1.93 $
+ *   $Revision: 1.94 $
  *
  *    Copyright (C) 2002 IBM Deutschland Entwicklung GmbH,
  *			 IBM Corporation
diff -urpN linux-2.6/drivers/s390/cio/device_ops.c linux-2.6-patched/drivers/s390/cio/device_ops.c
--- linux-2.6/drivers/s390/cio/device_ops.c	2006-01-03 04:21:10.000000000 +0100
+++ linux-2.6-patched/drivers/s390/cio/device_ops.c	2006-01-09 15:40:30.000000000 +0100
@@ -1,7 +1,7 @@
 /*
  *  drivers/s390/cio/device_ops.c
  *
- *   $Revision: 1.58 $
+ *   $Revision: 1.59.2.1 $
  *
  *    Copyright (C) 2002 IBM Deutschland Entwicklung GmbH,
  *			 IBM Corporation
diff -urpN linux-2.6/drivers/s390/cio/qdio.c linux-2.6-patched/drivers/s390/cio/qdio.c
--- linux-2.6/drivers/s390/cio/qdio.c	2006-01-09 15:40:30.000000000 +0100
+++ linux-2.6-patched/drivers/s390/cio/qdio.c	2006-01-09 15:40:30.000000000 +0100
@@ -56,7 +56,7 @@
 #include "ioasm.h"
 #include "chsc.h"
 
-#define VERSION_QDIO_C "$Revision: 1.114 $"
+#define VERSION_QDIO_C "$Revision: 1.115.2.1 $"
 
 /****************** MODULE PARAMETER VARIABLES ********************/
 MODULE_AUTHOR("Utz Bacher <utz.bacher@de.ibm.com>");
diff -urpN linux-2.6/drivers/s390/cio/qdio.h linux-2.6-patched/drivers/s390/cio/qdio.h
--- linux-2.6/drivers/s390/cio/qdio.h	2006-01-09 15:40:30.000000000 +0100
+++ linux-2.6-patched/drivers/s390/cio/qdio.h	2006-01-09 15:40:30.000000000 +0100
@@ -3,7 +3,7 @@
 
 #include "schid.h"
 
-#define VERSION_CIO_QDIO_H "$Revision: 1.40 $"
+#define VERSION_CIO_QDIO_H "$Revision: 1.41.2.1 $"
 
 #ifdef CONFIG_QDIO_DEBUG
 #define QDIO_VERBOSE_LEVEL 9
